pkg/agent/datamodel/types.go

```diff
@@ -2033,6 +2035,22 @@ type AKSKubeletConfiguration struct {
 	Default: []
 	+optional. */
 	AllowedUnsafeSysctls []string `json:"allowedUnsafeSysctls,omitempty"`
+	// shutdownGracePeriod specifies the total duration that the node should delay the
+	// shutdown and total grace period for pod termination during a node shutdown.
+	// Default: "0s"
+	// +featureGate=GracefulNodeShutdown
```

Comment:
we'll need to make sure that RP is setteing the `GracefulNodeShutdown` feature gate

Reply:
Feature gate is by default enabled in k8s 1.21+, and we can't edit nodes to have this config. Do we still need to enable the feature gate in this case?